src/components/ComputeModal.js

```diff
@@ -1012,6 +1082,33 @@ export const ComputeModalBase = ({ onDismiss, onSuccess, runtimes, persistentDis
     ])
   }
 
+  const renderDifferentLocationWarning = () => {
+    return div({ style: { ...computeStyles.drawerContent, ...computeStyles.warningView } }, [
+      h(TitleBar, {
+        id: titleId,
+        hideCloseButton: isAnalysisMode,
+        style: computeStyles.titleBar,
+        title: h(WarningTitle, ['Compute location differs from workspace bucket location']),
+        onDismiss,
+        onPrevious: () => setViewMode(undefined)
+      }),
+      div({ style: { lineHeight: 1.5 } }, [
+        p([
+          'This cloud environment will be created in the region ', computeConfig.computeRegion.toLowerCase(), '. ',
+          'Copying data from your workspace bucket in ', bucketLocation.toLowerCase(), ' may incur network egress charges.'
+        ]),
+        h(Link, { href: 'https://support.terra.bio/hc/en-us/articles/360058964552', ...Utils.newTabLinkProps }, [
+          'For more information please read the documentation.',
+          icon('pop-out', { size: 12, style: { marginLeft: '0.25rem' } })
+        ]),
+        p(['If you want your VM in ', computeConfig.computeRegion.toLowerCase(), ', continue. Otherwise, go back to select another location.'])
```

Comment:
Similarly as above

```suggestion
        p([`If you want your VM in ${computeConfig.computeRegion.toLowerCase()}, continue. Otherwise, go back to select another location.`])
```